lib/fastlane/plugin/wpmreleasetoolkit/actions/android/android_download_file_by_version.rb

```diff
@@ -0,0 +1,79 @@
+module Fastlane
+  module Actions
+    class AndroidDownloadFileByVersionAction < Action
+      def self.run(params)
+        require_relative '../../helper/android/android_localize_helper.rb'
+        require_relative '../../helper/github_helper'
+
+        version = Fastlane::Helper::Android::LocalizeHelper.get_library_version_from_gradle_config(import_key: params[:import_key])
+        UI.user_error!("Can't find any reference for key #{params[:import_key]}") if version.nil?
+        UI.message "Downloading #{params[:file_path]} from #{params[:repository]} at version #{version} to #{params[:download_folder]}"
+
+        Fastlane::Helper::GithubHelper.download_file_from_release(
+          repository: params[:repository],
+          release: "#{params[:github_release_prefix]}#{version}",
+          file_path: params[:file_path],
+          download_folder: params[:download_folder]
+        )
+      end
+
+      #####################################################
+      # @!group Documentation
+      #####################################################
+
+      def self.description
+        'Downloads a file from a GitHub release based on the version used by the client app'
+      end
+
+      def self.details
+        'This action extracts the version of the library which is imported by the client app' \
+        'and downloads the request file from the relevant GitHub release'
+      end
+
+      def self.available_options
+        [
+          FastlaneCore::ConfigItem.new(key: :library_name,
+                        description: "The display name of the library",
+                        type: String,
+                        optional: false),
+          FastlaneCore::ConfigItem.new(key: :import_key,
+                        description: "The key which is used in gradle.build to reference the version of the library to import",
+                        type: String,
+                        optional: false),
+          FastlaneCore::ConfigItem.new(key: :repository,
+                        description: "The GitHub repository which hosts the library project",
```

Comment:
The doc should tell what format is expected here. The full github URL to the repo? Just the repo slug like `org/reponame`?

```suggestion
                        description: "The GitHub repository slug ('<orgname>/<reponame>') which hosts the library project",
```